websockets/protocol.py

```diff
@@ -217,8 +219,8 @@ def connection_open(self):
 
         """
         # 4.1. The WebSocket Connection is Established.
-        assert self.state == CONNECTING
-        self.state = OPEN
+        assert self.state == State.CONNECTING
```

Comment:
The doc says to use `is` to compare enums. It was recently mentioned to me and it didn't really come naturally to me. I guess it sort of makes sense if one wants to force the use of enums and forgo the `int` capability (since `State` is an `IntEnum`).

Anyway, I'm not 100% sold on this whole `is` thing, but I thought I'd mention it and see what you guys think.

Reply:
That's... unsettling. But why not. Let's make the change if that's the best practice. Perhaps it's faster.

Reply:
Fixed in f730fe091fdb9aab8b608e710d260efdf4e8a077.